contracts/ERC721R.sol

```diff
@@ -315,20 +315,22 @@ contract ERC721r is Context, ERC165, IERC721, IERC721Metadata {
         }
 
         uint256 lastIndex = updatedNumAvailableTokens - 1;
+        uint256 lastValInArray = _availableTokens[lastIndex];
         if (indexToUse != lastIndex) {
             // Replace the value at indexToUse, now that it's been used.
             // Replace it with the data from the last index in the array, since we are going to decrease the array size afterwards.
-            uint256 lastValInArray = _availableTokens[lastIndex];
             if (lastValInArray == 0) {
                 // This means the index itself is still an available token
                 _availableTokens[indexToUse] = lastIndex;
             } else {
                 // This means the index itself is not an available token, but the val at that index is.
                 _availableTokens[indexToUse] = lastValInArray;
-                // Gas refund courtsey of @dievardump
-                delete _availableTokens[lastIndex];
             }
         }
+        if (lastValInArray != 0) {
```

Comment:
Minor question: do you happen to know if we need this if? What if we just try to delete it in any case, is that worse for gas?

Reply:
I'm not sure.
Could you do some experiments with the following code or something?

```solidity
contract TestDeleteMapping {
    mapping (uint256 => uint256) map;

    function write() external {
        map[1] = 1;
    }

    function del() external {
        delete map[1];
    }

    function delIfNotEmpty() external {
        if (map[1] != 0) {
            delete map[1];
        }
    }
}
```

Even if deleting empty entries is allowed,
I think it's always better we keep the `if` statement.
It makes the code more explicit.

Reply:
Makes sense! Looks good to me!